src/cargo/ops/cargo_package/vcs.rs

```diff
@@ -133,12 +134,14 @@ fn git(
     // Find the intersection of dirty in git, and the src_files that would
     // be packaged. This is a lazy n^2 check, but seems fine with
     // thousands of files.
+    let mut dirty_files_outside_pkg_root = dirty_symlinks(pkg, repo, src_files)?;
+    dirty_files_outside_pkg_root.extend(dirty_metadata_paths(pkg, repo)?);
     let cwd = gctx.cwd();
     let mut dirty_src_files: Vec<_> = src_files
         .iter()
         .filter(|src_file| dirty_files.iter().any(|path| src_file.starts_with(path)))
         .map(|p| p.as_ref())
-        .chain(dirty_metadata_paths(pkg, repo)?.iter())
+        .chain(dirty_files_outside_pkg_root.iter())
```

Comment:
There a reason you aren't chaining them?

Reply:
Deduplicate same symlinks both reported in `dirty_metadata_paths` and `dirty_symlinks`.

I would go either

* make `dirty_metadata_paths` stop handling symlinks.
* make this a separate commit to show the dedup change in Git history
* combine `dirty_metadata_paths` and `dirty_symlinks` into a single function.

Any preference?

Reply:
In my C++ days, I would use an out-param instead of every function generating a data structure that allocates.  That approach isn't in too much favor in the Rust community though.

What we have is probably fine.

Reply:
I've done one additional commit to make it clear that we dedup.